src/os/bluestore/BlueStore.cc

```diff
@@ -1465,6 +1465,8 @@ void BlueStore::_set_compression()
     }
   }
   comp_mode = m;
+  if (comp_mode == COMP_NONE)
```

Comment:
Probably it's better to avoid compressor allocation at all if mode is inappropriate. I.e. calculate the mode first and then proceed with allocation